doc/api/metrics.md

```diff
@@ -0,0 +1,1040 @@
+
+apiCalls.increment();     // Increment by 1
+apiCalls.increment(5);    // Increment by 5
+apiCalls.increment(10, { endpoint: '/api/users' }); // Increment by 10 with metadata
+apiCalls.increment({ endpoint: '/api/orders' });    // Increment by 1 with metadata
+```
+
+#### `counter.decrement([n[, meta]])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `n` {number} The amount to decrement. **Default:** `1`
+* `meta` {Object} Additional metadata for this report.
+
+Decrements the counter by the specified amount.
+
+```mjs
+import { counter } from 'node:metrics';
+
+const errorCount = counter('errors.total', { component: 'database' });
+
+errorCount.decrement();     // Decrement by 1
+errorCount.decrement(3);    // Decrement by 3
+errorCount.decrement(2, { errorType: 'timeout' }); // Decrement by 2 with metadata
+errorCount.decrement({ errorType: 'timeout' });    // Decrement by 1 with metadata
+```
+
+#### `counter.value`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The current value of the counter.
+
+### Class: `Gauge`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+A metric representing a single value that can go up or down.
+
+#### `gauge.metric`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {metrics.Metric}
+
+The underlying metric instance used for reporting.
+
+#### `gauge.value`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {number}
+
+The current value of the metric.
+
+#### `gauge.reset([value[, meta]])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `value` {number} The new value. **Default:** `0`
+* `meta` {Object} Additional metadata for this report.
+
+Sets the gauge to a specific value and reports it.
+
+```mjs
+import { gauge } from 'node:metrics';
+import { memoryUsage } from 'node:process';
+
+const memory = gauge('memory.usage.bytes');
+
+memory.reset(); // Reset to 0
+memory.reset(memoryUsage().heapUsed); // Set to current memory usage
+memory.reset(1024, { source: 'system' }); // Set to 1024 with metadata
+```
+
+#### `gauge.applyDelta(delta[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `delta` {number} The amount to add to the current value.
+* `meta` {Object} Additional metadata for this report.
+
+Adds a delta to the current value and reports the new value.
+
+```mjs
+import { gauge } from 'node:metrics';
+
+const cpuUsage = gauge('cpu.usage.percent');
+
+cpuUsage.applyDelta(5); // Increase by 5
+cpuUsage.applyDelta(-2, { source: 'system' }); // Decrease by 2 with metadata
+```
+
+### Class: `Meter`
+
+* Extends: {metrics.Gauge}
+
+<!-- YAML
+added: REPLACEME
+-->
+
+A metric that measures the rate of events over a sliding time window.
+
+#### `meter.metric`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* {metrics.Metric}
+
+The underlying metric instance used for reporting.
+
+#### `meter.mark([n[, meta]])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `n` {number} The number of events to mark. **Default:** `1`
+* `meta` {Object} Additional metadata for this report.
+
+Records events and updates the rate calculation.
```

Comment:
Having a meter of marks per time interval is great but I wonder if it would be worthwhile to have a meter of marks per other unit as well. For instance, something like...

```
const rate = createMeter('marks.per.tick');
rate.mark(10);
rate.tick();  // 10 marks / tick
rate.mark(20);
rate.tick(); // 20 marks / tick
// etc
```